### PR TITLE
SAK-33102 copy sort order in GBNG

### PR DESCRIPTION
**Pull request by RyanAFinney**

### CODE REVIEW

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/InstructorGradeSummaryGradesPanel.java [hunk 1]
```diff
@@ -107,22 +112,55 @@ public void onBeforeRender() {
...
+		if(this.categoriesEnabled){
+			//sort assignments in each category
+			categoryNamesToAssignments.forEach((categoryName, assignmentsInCategory) -> {
+				Collections.sort(assignmentsInCategory);
+			});
+
+			//sort categories, if needed
+			List<Integer> sortedCategoryKeys = new ArrayList<>(sortableCategoryNames.keySet());
+			//null safe integer sort forcing uncategorized to bottom of list
+			Collections.sort(sortedCategoryKeys, (order1, order2) -> {
```
Comment (jonespm) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/InstructorGradeSummaryGradesPanel.java]: I think you can just use nullsFirst for this? https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#nullsFirst-java.util.Comparator-

**Comment by ern** [comment]
I feel like this PR could be simplified by creating a comparator and sort using that vs creating a temporary collection.

**Comment by payten** [comment]
Hi all, 

To sort the grade matrix for the table, we push this to the `GradebookService` by passing a `SortType` to `GradebookNgBusinessService.getGradebookAssignments(final String siteId, final SortType sortBy)`.  This saves doing any extra sorting in the tool.  I think you could use this to save any extra sorting headaches.

Originally, the instructor view had slightly different sorting requirements to the big table where instead we sorted by due date and then the assignment name (perhaps to match up with the Gradebook Classic?).  This is largely facilitated by JavaScript via the tablesorter plugin.

So to get things working, we'll need to disable those tablesorter settings and pass a `SortType` through to the `GradebookService`.  I just had a quick play and it looks like this is possible by introducing a new `GradebookNgBusinessService` method `getGradebookAssignmentsForStudent(final String studentUuid, final SortType sortedBy)`: https://github.com/payten/sakai/commit/cdfb9508aaba3151b626ea85442d5066622bfd28.

Hopefully that's helpful.

Thanks,
Payten

**Comment by davidpbauer** [comment]
Maybe I'm missing something with this feature, but the commit @payten linked to seems to work fine. I've reverted Ryan's commit and cherry-picked Payten's. Let me know if you want me to do anything with this one.

**Pull request merged**